src/signature.js

```diff
@@ -0,0 +1,128 @@
+import crypto from "crypto";
+import fetch from "node-fetch";
+
+import { account, domain } from "./util.js";
+import { getPrivateKey } from "./activity-pub-db.js";
```

Comment:
@ckolderup Calling this out as something I was a little unsure of.

Previously, the route functions would pass the account name, domain, and db to functions like `signAndSend()` in order to sign requests. Initially, I was just passing them through to this new `signedFetch()` function, which was a little clunky but not too bad.

However, continuing to do it that way would mean that `getInboxFromActorProfile()` (and any other functions making unsigned GET requests) would need to be updated to receive those values as well, just to pass them through to `signedFetch()`.

It seemed much simpler to grab the values directly rather than passing them from the route through multiple other function calls. So that's what I did.

But maybe there was a really good architectural or security reason NOT to do it that way? I have no idea. Let me know what you think.

Reply:
it's mostly an artifact of this codebase evolving over a series of individual hacking sessions. I think my instincts are to pass more things because I've worked in a lot of DI-heavy codebases recently and it is kinda nice that functions are easier to test as a result, but for better or worse the testing hasn't even begun on this codebase (if I can find a large block of time to sit down soon that's my top priority to put foundations in place for), so there's no sense in making your code harder to understand in the meantime.